logstreams/src/main/java/io/zeebe/logstreams/state/StateSnapshotController.java

```diff
@@ -60,7 +60,8 @@ public StateSnapshotController(
                 LOG.error("Unexpected error occurred on ensuring max snapshot count.", ioe);
               }
             },
-            () -> getPositionToDelete(maxSnapshots));
+            () -> getPositionToDelete(maxSnapshots),
```

Comment:
Might make more sense to have a callback which is triggered on receiving a valid snapshot. In this callback we then use our information to delete stuff, instead of have multiple suppliers and the DeletionService in the ReplicationController.